central/pruning/pruning.go

```diff
@@ -485,6 +494,127 @@ func (g *garbageCollectorImpl) collectImages(config *storage.PrivateConfig) {
 	}
 }
 
+func (g *garbageCollectorImpl) collectClusters(config *storage.PrivateConfig) {
+	if !features.DecommissionedClusterRetention.Enabled() {
+		return
+	}
+
+	if lastClusterPruneTime.Add(clusterGCFreq).After(time.Now()) {
+		// Only cluster pruning if it's been at least clusterGCFreq since last run
+		return
+	}
+	defer func() {
+		lastClusterPruneTime = time.Now()
+	}()
+
+	clusterRetention := config.GetDecommissionedClusterRetention()
+	retentionDays := int64(clusterRetention.GetRetentionDurationDays())
+	if retentionDays == 0 {
```

Comment:
I realized that our retention configuration is treated inconsistently. Zero value for the images means no retention (because of AddDays query)  and immediate GC whereas for alerts pruning means no GC. (@viswajithiii ?)

We should just make the retention consistent for zero value (meaning disabled).

Reply:
Agreed on that. However it's a backwards incompatible change, so let's tackle is separately from this project?